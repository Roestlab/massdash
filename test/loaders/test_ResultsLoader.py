"""
test/loaders/test_ResultsLoader.py
"""

from pathlib import Path
import pytest
import pandas as pd
from syrupy.extensions.amber import AmberDataSerializer

from massdash.loaders import ResultsLoader
from massdash.util import find_git_directory
from massdash.testing import PandasSnapshotExtension
from massdash.loaders.access.OSWDataAccess import OSWDataAccess

TEST_PATH = find_git_directory(Path(__file__).resolve()).parent / 'test'

@pytest.fixture
def snapshot_pandas(snapshot):
    return snapshot.use_extension(PandasSnapshotExtension)

#f"{TEST_PATH}/test_data/example_dia/dreamdia/test_dreamdia_report.tsv",
    #[f"{TEST_PATH}/test_data/example_dia/diann/report/test_diann_report_combined.tsv", f"{TEST_PATH}/test_data/example_dia/dreamdia/test_dreamdia_report.tsv"]
    # NOTE: with this PR dreamDIA is not supported
@pytest.fixture(params=['openswath', 'diann1', 'combined'])
def resultsLoader(request):
    if request.param == 'openswath':
        return ResultsLoader(rsltsFile=f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw", libraryFile=None, verbose=False, mode='module')
    elif request.param == 'diann1':
        return ResultsLoader(rsltsFile=f"{TEST_PATH}/test_data/example_dia/diann/report/test_1_diann_report.tsv", libraryFile=None, verbose=False, mode='module')
    elif request.param == 'combined':
        return ResultsLoader(rsltsFile=[f"{TEST_PATH}/test_data/example_dia/diann/report/test_diann_report_combined.tsv", 
                                        f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw"], libraryFile=None, verbose=False, mode='module')
    else:
        raise ValueError(f"Invalid parameter: {request.param}")

@pytest.fixture
def oswResultsLoader():
    return ResultsLoader(rsltsFile=f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw", libraryFile=None, verbose=False, mode='module')

@pytest.fixture(params=['AGAANIVPNSTGAAK', 'INVALID'])
def precursor(request):
    return request.param

@pytest.fixture(params=[2])
def charge(request):
    return request.param

def test_loadTransitionGroupFeatures(resultsLoader, precursor, charge, snapshot):
    features = resultsLoader.loadTransitionGroupFeatures(precursor, charge)
    assert snapshot == AmberDataSerializer.serialize(features)

def test_loadTopTransitionGroupFeature(resultsLoader, precursor, charge, snapshot):
    top_feature = resultsLoader.loadTopTransitionGroupFeature(precursor, charge)
    assert snapshot == AmberDataSerializer.serialize(top_feature)

def test_loadTopTransitionGroupFeatureDf(resultsLoader, precursor, charge, snapshot_pandas):
    top_feature = resultsLoader.loadTopTransitionGroupFeatureDf(precursor, charge)
    top_feature = top_feature.sort_values(by=['consensusApex'], ascending=True).reset_index(drop=True)
    assert snapshot_pandas == top_feature

def test_loadTransitionGroupFeaturesDf(resultsLoader, precursor, charge, snapshot_pandas):
    feature = resultsLoader.loadTransitionGroupFeaturesDf(precursor, charge)
    feature = feature.sort_values(by=['consensusApex'], ascending=True).reset_index(drop=True)
    assert snapshot_pandas == feature

@pytest.mark.parametrize('run', ['test_raw_1', ['test_raw_1', 'test_raw_2'], 'test_raw_3']) #note that 'test_raw_3' is not in the data
def test_loadTransitionGroupFeaturesDf_runSpecific(resultsLoader, precursor, charge, run, snapshot_pandas):
    feature = resultsLoader.loadTransitionGroupFeaturesDf(precursor, charge, runNames=run)
    print(feature)
    feature = feature.sort_values(by=['consensusApex'], ascending=True).reset_index(drop=True)
    assert snapshot_pandas == feature

def test_loadSoftware(resultsLoader, snapshot):
    assert snapshot == resultsLoader._loadSoftware()

def test_inferRunNames(resultsLoader, snapshot):
    assert snapshot == resultsLoader._inferRunNames()

def test_loadIdentifiedPrecursors(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadIdentifiedPrecursors()

def test_loadNumIdentifiedPrecursors(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadNumIdentifiedPrecursors()

def test_loadIdentifiedProteins(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadIdentifiedProteins()

def test_loadNumIdentifiedProteins(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadNumIdentifiedProteins()

def test_loadIdentifiedPeptides(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadIdentifiedPeptides()

def test_loadNumIdentifiedPeptides(resultsLoader, snapshot):
    assert snapshot == resultsLoader.loadNumIdentifiedPeptides()

def test_loadExperimentSummary(resultsLoader, snapshot_pandas):
    assert snapshot_pandas == resultsLoader.loadExperimentSummary()

def test_loadQuantificationMatrix(resultsLoader, snapshot_pandas):
    assert snapshot_pandas == resultsLoader.loadQuantificationMatrix()

def test_computeCV(resultsLoader, snapshot_pandas):
    assert snapshot_pandas == resultsLoader.computeCV()

@pytest.mark.parametrize('rsltsFiles,expected', (
        [f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw", OSWDataAccess],
        [f"{TEST_PATH}/test_data/example_dia/diann/report/test_1_diann_report.tsv", None],
        [[f"{TEST_PATH}/test_data/example_dia/diann/report/test_diann_report_combined.tsv", f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw"], OSWDataAccess],
        [[f"{TEST_PATH}/test_data/osw/ionMobilityTest.osw", f"{TEST_PATH}/test_data/example_dia/openswath/osw/test.osw"], None] # multiple osw files
)  )
def test_getOSWAccessPtr(rsltsFiles, expected):
    resultsLoader = ResultsLoader(rsltsFile=rsltsFiles, libraryFile=None, verbose=False, mode='module')
    if expected is None:
        assert expected is resultsLoader.getOSWAccessPtr()
    else: # expected is OSWDataAccess
        assert isinstance(resultsLoader.getOSWAccessPtr(), expected)

def test_loadPeakGroupScoreDistribution(oswResultsLoader, snapshot_pandas):
    assert snapshot_pandas == oswResultsLoader.loadPeakGroupScoreDistribution()

@pytest.mark.parametrize('context', ('global', 'run-specific', 'experiment-wide'))
def test_loadPeptideScoreDistribution(oswResultsLoader, context, snapshot_pandas):
    assert snapshot_pandas == oswResultsLoader.loadPeptideScoreDistribution(context=context)

@pytest.mark.parametrize('context', ('global', 'run-specific', 'experiment-wide'))
def test_loadProteinScoreDistribution(oswResultsLoader, context, snapshot_pandas):
    assert snapshot_pandas == oswResultsLoader.loadProteinScoreDistribution(context=context)

def test_loadScoreDistribution(oswResultsLoader, snapshot_pandas):
    pass # This just invokes getScoreTable from OSWDataAccess if getOSWAccessPtr is working properly this should work

def test_flattenDict():
    arg_in = {'a': {'a1': 1, 'a2': 2}, 'b': {'b1' : 1, 'b2': 2} }
    arg_out = ResultsLoader._flattenDict(arg_in)
    print(arg_out)
    assert arg_out == {'a1 (a)': 1, 'a2 (a)': 2, 'b1 (b)': 1, 'b2 (b)': 2}

'''
def test_plotIdentifications(resultsLoader, snapshot_bokeh):
    assert snapshot_bokeh == resultsLoader.plotIdentifications()

def test_plotQuantification(resultsLoader, snapshot_bokeh):
    assert snapshot_bokeh == resultsLoader.plotQuantification()

def test_plotCV(resultsLoader, snapshot_bokeh):
    assert snapshot_bokeh == resultsLoader.plotCV()

def test_plotUpset(resultsLoader, snapshot_bokeh):
    assert snapshot_bokeh == resultsLoader.plotUpset()
'''


