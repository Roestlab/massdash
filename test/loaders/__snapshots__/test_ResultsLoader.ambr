# serializer version: 1
# name: test_inferRunNames[combined]
  list([
    'test_raw_1',
    'test_raw_2',
  ])
# ---
# name: test_inferRunNames[diann1]
  list([
    'test_raw_1',
  ])
# ---
# name: test_inferRunNames[openswath]
  list([
    'test_raw_1',
    'test_raw_2',
  ])
# ---
# name: test_loadIdentifiedPeptides[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
    }),
    'OpenSWATH': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedPeptides[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedPeptides[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK',
        'DYASIDAAPEER',
        'IEDQLGEVAQYK',
        'KVVITAPGGNDVK',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedPrecursors[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
    }),
    'OpenSWATH': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedPrecursors[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedPrecursors[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
      'test_raw_2': set({
        'AGAANIVPNSTGAAK2',
        'AGAANIVPNSTGAAK3',
        'DYASIDAAPEER2',
        'DYASIDAAPEER3',
        'IEDQLGEVAQYK2',
        'IEDQLGEVAQYK3',
        'KVVITAPGGNDVK2',
        'KVVITAPGGNDVK3',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedProteins[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'P0C0G6;P0C0G7;P0DB18;Q5XDW3;P0DB19;P68777',
        'Q5XD01;P69949;A2RFE3;B5XKM7;P0DA94;P0DA95;P69951;Q1J7I5;Q1JCN8;Q1JHQ6;Q1JML5;Q48UF7',
        'Q5XD49;A2RFQ4;B5XKI1;P0DA82;P0DA83;P69952;Q1J7N4;Q1JCT6;Q1JHV6;Q1JMR3;Q48UK5;Q8P1W4',
      }),
      'test_raw_2': set({
        'P0C0G6;P0C0G7;P0DB18;Q5XDW3;P0DB19;P68777',
        'Q5XD01;P69949;A2RFE3;B5XKM7;P0DA94;P0DA95;P69951;Q1J7I5;Q1JCN8;Q1JHQ6;Q1JML5;Q48UF7',
        'Q5XD49;A2RFQ4;B5XKI1;P0DA82;P0DA83;P69952;Q1J7N4;Q1JCT6;Q1JHV6;Q1JMR3;Q48UK5;Q8P1W4',
      }),
    }),
    'OpenSWATH': dict({
      'test_raw_1': set({
        'Spyo_Exp3652_DDB_SeqID_514830',
        'Spyo_Exp3652_DDB_SeqID_514851',
        'Spyo_Exp3652_DDB_SeqID_515131',
      }),
      'test_raw_2': set({
        'Spyo_Exp3652_DDB_SeqID_514830',
        'Spyo_Exp3652_DDB_SeqID_514851',
        'Spyo_Exp3652_DDB_SeqID_515131',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedProteins[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': set({
        'P0C0G6;P0C0G7;P0DB18;Q5XDW3;P0DB19;P68777',
        'Q5XD01;P69949;A2RFE3;B5XKM7;P0DA94;P0DA95;P69951;Q1J7I5;Q1JCN8;Q1JHQ6;Q1JML5;Q48UF7',
        'Q5XD49;A2RFQ4;B5XKI1;P0DA82;P0DA83;P69952;Q1J7N4;Q1JCT6;Q1JHV6;Q1JMR3;Q48UK5;Q8P1W4',
      }),
    }),
  })
# ---
# name: test_loadIdentifiedProteins[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': set({
        'Spyo_Exp3652_DDB_SeqID_514830',
        'Spyo_Exp3652_DDB_SeqID_514851',
        'Spyo_Exp3652_DDB_SeqID_515131',
      }),
      'test_raw_2': set({
        'Spyo_Exp3652_DDB_SeqID_514830',
        'Spyo_Exp3652_DDB_SeqID_514851',
        'Spyo_Exp3652_DDB_SeqID_515131',
      }),
    }),
  })
# ---
# name: test_loadNumIdentifiedPeptides[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 4,
      'test_raw_2': 4,
    }),
    'OpenSWATH': dict({
      'test_raw_1': 4,
      'test_raw_2': 4,
    }),
  })
# ---
# name: test_loadNumIdentifiedPeptides[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 4,
    }),
  })
# ---
# name: test_loadNumIdentifiedPeptides[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': 4,
      'test_raw_2': 4,
    }),
  })
# ---
# name: test_loadNumIdentifiedPrecursors[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 8,
      'test_raw_2': 8,
    }),
    'OpenSWATH': dict({
      'test_raw_1': 8,
      'test_raw_2': 8,
    }),
  })
# ---
# name: test_loadNumIdentifiedPrecursors[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 8,
    }),
  })
# ---
# name: test_loadNumIdentifiedPrecursors[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': 8,
      'test_raw_2': 8,
    }),
  })
# ---
# name: test_loadNumIdentifiedProteins[combined]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 3,
      'test_raw_2': 3,
    }),
    'OpenSWATH': dict({
      'test_raw_1': 3,
      'test_raw_2': 3,
    }),
  })
# ---
# name: test_loadNumIdentifiedProteins[diann1]
  dict({
    'DIA-NN': dict({
      'test_raw_1': 3,
    }),
  })
# ---
# name: test_loadNumIdentifiedProteins[openswath]
  dict({
    'OpenSWATH': dict({
      'test_raw_1': 3,
      'test_raw_2': 3,
    }),
  })
# ---
# name: test_loadSoftware[combined]
  list([
    'DIA-NN',
    'OpenSWATH',
  ])
# ---
# name: test_loadSoftware[diann1]
  list([
    'DIA-NN',
  ])
# ---
# name: test_loadSoftware[openswath]
  list([
    'OpenSWATH',
  ])
# ---
# name: test_loadTopTransitionGroupFeature[combined-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1933.4520000000002
  rightBoundary: 1967.5919999999999
  areaIntensity: 10209.3
  consensusApex: 1950.522
  consensusApexIntensity: None
  qvalue: 2.90751e-06
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
      -------- TransitionGroupFeature --------
  leftBoundary: 1923.2130126953125
  rightBoundary: 2001.72900390625
  areaIntensity: 4390.32825839279
  consensusApex: 1953.14
  consensusApexIntensity: 4390.32825839279
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
    'test_raw_2': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1912.974
  rightBoundary: 1940.2859999999998
  areaIntensity: 9815.24
  consensusApex: 1926.63
  consensusApexIntensity: None
  qvalue: 2.84659e-05
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
      -------- TransitionGroupFeature --------
  leftBoundary: 1895.906005859375
  rightBoundary: 1971.010986328125
  areaIntensity: 2540.8210337067203
  consensusApex: 1920.24
  consensusApexIntensity: 2540.8210337067203
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
  })
  '''
# ---
# name: test_loadTopTransitionGroupFeature[combined-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
    'test_raw_2': list([
    ]),
  })
  '''
# ---
# name: test_loadTopTransitionGroupFeature[diann1-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1933.4520000000002
  rightBoundary: 1967.5919999999999
  areaIntensity: 10209.3
  consensusApex: 1950.522
  consensusApexIntensity: None
  qvalue: 2.90751e-06
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
    ]),
  })
  '''
# ---
# name: test_loadTopTransitionGroupFeature[diann1-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
  })
  '''
# ---
# name: test_loadTopTransitionGroupFeature[openswath-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1923.2130126953125
  rightBoundary: 2001.72900390625
  areaIntensity: 4390.32825839279
  consensusApex: 1953.14
  consensusApexIntensity: 4390.32825839279
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
    'test_raw_2': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1895.906005859375
  rightBoundary: 1971.010986328125
  areaIntensity: 2540.8210337067203
  consensusApex: 1920.24
  consensusApexIntensity: 2540.8210337067203
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
  })
  '''
# ---
# name: test_loadTopTransitionGroupFeature[openswath-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
    'test_raw_2': list([
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[combined-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1933.4520000000002
  rightBoundary: 1967.5919999999999
  areaIntensity: 10209.3
  consensusApex: 1950.522
  consensusApexIntensity: None
  qvalue: 2.90751e-06
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
      -------- TransitionGroupFeature --------
  leftBoundary: 2206.55908203125
  rightBoundary: 2250.93701171875
  areaIntensity: 93.7645
  consensusApex: 2226.53
  consensusApexIntensity: 25.361407994046484
  qvalue: 0.27997408312165795
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2001.72900390625
  rightBoundary: 2039.2840576171875
  areaIntensity: 70.6678
  consensusApex: 2009.74
  consensusApexIntensity: 33.88131947814447
  qvalue: 1.822896508451022e-05
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2158.76611328125
  rightBoundary: 2203.14501953125
  areaIntensity: 113.104
  consensusApex: 2177.07
  consensusApexIntensity: 30.8382840089739
  qvalue: 0.16741273086102035
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2117.801025390625
  rightBoundary: 2158.76611328125
  areaIntensity: 784.279
  consensusApex: 2138.98
  consensusApexIntensity: 165.07510208344246
  qvalue: 0.0009590506619916316
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1923.2130126953125
  rightBoundary: 2001.72900390625
  areaIntensity: 44921.8
  consensusApex: 1953.14
  consensusApexIntensity: 4390.32825839279
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
    'test_raw_2': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1912.974
  rightBoundary: 1940.2859999999998
  areaIntensity: 9815.24
  consensusApex: 1926.63
  consensusApexIntensity: None
  qvalue: 2.84659e-05
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
      -------- TransitionGroupFeature --------
  leftBoundary: 2104.14990234375
  rightBoundary: 2145.115966796875
  areaIntensity: 417.457
  consensusApex: 2126.42
  consensusApexIntensity: 95.16263446243194
  qvalue: 0.0004287466278223515
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1971.010986328125
  rightBoundary: 2039.2860107421875
  areaIntensity: 1049.18
  consensusApex: 2000.31
  consensusApexIntensity: 159.3757949554613
  qvalue: 3.35632586944344e-05
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1895.906005859375
  rightBoundary: 1971.010986328125
  areaIntensity: 30771.2
  consensusApex: 1920.24
  consensusApexIntensity: 2540.8210337067203
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2145.115966796875
  rightBoundary: 2182.6708984375
  areaIntensity: 60.9997
  consensusApex: 2162.68
  consensusApexIntensity: 15.554447237443734
  qvalue: 0.38296484850166657
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[combined-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
    'test_raw_2': list([
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[diann1-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 1933.4520000000002
  rightBoundary: 1967.5919999999999
  areaIntensity: 10209.3
  consensusApex: 1950.522
  consensusApexIntensity: None
  qvalue: 2.90751e-06
  consensusApexIM: 0
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: DIA-NN,
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[diann1-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[openswath-AGAANIVPNSTGAAK-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 2206.55908203125
  rightBoundary: 2250.93701171875
  areaIntensity: 93.7645
  consensusApex: 2226.53
  consensusApexIntensity: 25.361407994046484
  qvalue: 0.27997408312165795
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2001.72900390625
  rightBoundary: 2039.2840576171875
  areaIntensity: 70.6678
  consensusApex: 2009.74
  consensusApexIntensity: 33.88131947814447
  qvalue: 1.822896508451022e-05
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2158.76611328125
  rightBoundary: 2203.14501953125
  areaIntensity: 113.104
  consensusApex: 2177.07
  consensusApexIntensity: 30.8382840089739
  qvalue: 0.16741273086102035
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2117.801025390625
  rightBoundary: 2158.76611328125
  areaIntensity: 784.279
  consensusApex: 2138.98
  consensusApexIntensity: 165.07510208344246
  qvalue: 0.0009590506619916316
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1923.2130126953125
  rightBoundary: 2001.72900390625
  areaIntensity: 44921.8
  consensusApex: 1953.14
  consensusApexIntensity: 4390.32825839279
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
    'test_raw_2': list([
      -------- TransitionGroupFeature --------
  leftBoundary: 2104.14990234375
  rightBoundary: 2145.115966796875
  areaIntensity: 417.457
  consensusApex: 2126.42
  consensusApexIntensity: 95.16263446243194
  qvalue: 0.0004287466278223515
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1971.010986328125
  rightBoundary: 2039.2860107421875
  areaIntensity: 1049.18
  consensusApex: 2000.31
  consensusApexIntensity: 159.3757949554613
  qvalue: 3.35632586944344e-05
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 1895.906005859375
  rightBoundary: 1971.010986328125
  areaIntensity: 30771.2
  consensusApex: 1920.24
  consensusApexIntensity: 2540.8210337067203
  qvalue: 6.993573006436113e-06
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
      -------- TransitionGroupFeature --------
  leftBoundary: 2145.115966796875
  rightBoundary: 2182.6708984375
  areaIntensity: 60.9997
  consensusApex: 2162.68
  consensusApexIntensity: 15.554447237443734
  qvalue: 0.38296484850166657
  consensusApexIM: None
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AGAANIVPNSTGAAK
  software: OpenSWATH,
    ]),
  })
  '''
# ---
# name: test_loadTransitionGroupFeatures[openswath-INVALID-2]
  '''
  TransitionGroupFeatureCollection({
    'test_raw_1': list([
    ]),
    'test_raw_2': list([
    ]),
  })
  '''
# ---
